### DISCUSSION TITLE
Actions when selecting multiple entries in list

### DISCUSSION THREAD

**Discussion by patrickunterwegs** [TechbeeAT/jtxBoard#91] [Ideas]
Hey @Djey51, hey @rogercreagh, 
we have been discussing this feature on GitLab, so if you're interested, we can talk about it here 😉 

Basically selecting entries in the list and deleting the entries is already implemented. Additionally I have removed the "Delete all visible" option, as it can easily be done by pushing the select all button.

But now I'm unsure what other actions could be implemented.
Categories yes. But I guess the process requires also to set a category, unset/delete a category and handling the possibility, that some entries might have and might not have the category already assigned. 

But what about any other attributes? 
I'm thinking about the resources, but I doubt that a lot of people use it. 
Updating dates? Is there a valid usecase for that?
Any other attributes?

I'm thinking about limiting the functionality for now to the categories. But maybe we can have a short discussion about it. 

Here's the current state of the development and how it looks like: 

![Screenshot_1672135964](https://user-images.githubusercontent.com/72232737/209659784-c289e182-18ba-4672-a219-400d23a8a3a6.png)

(also the symbol is something that I'm not sure about...)

![Screenshot_1672135986](https://user-images.githubusercontent.com/72232737/209659812-84522d37-b432-4511-ad72-2ea107c3f1a8.png)

For the folder-icon I thought about letting the entries be moved to another collection, but I'm also not sure about it.

So I'd appreciate your feedback 😉

**Comment by rogercreagh** [comment]
Personally I can see two use-cases for wanting to be able to make the same change to multiple entries:

1. Assigning categories/tags
  1a. Assigning an additional tag to several items (ignoring ones that already have the tag assigned in case of erroneous selection so as not to have it doubly assigned if that is even possible in the datastructure) 
  1b. Removing a tag from several items (again skipping any that don't have the tag rather than throwing an error)

2. Changing collections - this would be useful occasionally - eg I am considering switching to using a new collection for each year (rather than have over 365 entries in a collection)
  2a. Move selected items to a different collection
  2b Copy selected items to a different collection leaving them in place in the original (especially for the paranoid user who wants to check that all is well in the new collection before deleting them from the old!)

In the case of 1. it might be nice to assign/remove multiple tags at once, but not essential as you'd seldom want to do more than 2 or 3 and could do them sequentially.

I guess there might also be a case for batch changing status and classification but I don't currently use those.

I suppose for remove tags (categories) a second box-arrow (label? tag?) icon could have a minus sign in it, or add/remove could be an option on the screen where you select the category. I'm assuming having selected some items in the list pressing the box arrow will bring up a screen to chose a category and a confirmation button (do-it or cancel). This could also have a remove/assign choice.

Currently I have about a dozen tags in use, but I can imagine some instances might have many many more - the same selection dialog as when initially assigning categories would be ok. For 'remove tag' the list could be filtered to show only the tags currently assigned to any of the selected items. As I said above limiting this to adding/removing just one tag at a time would be fine if it makes things simpler

Likewise for copy/move items to different collection (folder icon is ok) presumably the next screen when pressing the folder button will offer a list of collections and an option to select one and a confirm/cancel button(s).

In the case of tag changes will the selection remain once the assign/remove process is done - would be handy if you did want to do another tag after the first.
